packages/@aws-cdk/aws-appsync/lib/graphqlapi-base.ts

```diff
@@ -0,0 +1,189 @@
+import { ITable } from '@aws-cdk/aws-dynamodb';
+import { IFunction } from '@aws-cdk/aws-lambda';
+import { CfnResource, IResource, Resource } from '@aws-cdk/core';
+import { DynamoDbDataSource, HttpDataSource, LambdaDataSource, NoneDataSource } from './data-source';
+
+/**
+ * Optional configuration for data sources
+ */
+export interface DataSourceOptions {
+  /**
+   * The name of the data source
+   *
+   * @default - '<DataSourceType>CDKDefault'
```

Comment:
perhaps rephrase to`Automatically generated name` and include an example.
I'm not super fond of the default part but I agree CDK should be in the default.

can multiple data sources be added with the same "friendly name" or will that complain? Can I just add 2 data sources of the same type with the defaults?

Reply:
No, i created a unit test to make sure of this. But essentially each resolver is a construct and a construct must have a unique name.